external/vcm/vcm/limit.py

```diff
@@ -0,0 +1,99 @@
+import xarray as xr
+from sklearn.base import BaseEstimator, TransformerMixin
```

Comment:
I don't think it's worth adding the direct scikit-learn dep to `runtime` just for these base classes. Also, sklearn classes usually take in and return numpy arrays. 

Duck typing should be sufficient here, and we can define our own protocol if we extend the method.